pkg/ccl/changefeedccl/protected_timestamps.go

```diff
@@ -31,14 +34,50 @@ func createProtectedTimestampRecord(
 ) *ptpb.Record {
 	ptsID := uuid.MakeV4()
 	deprecatedSpansToProtect := makeSpansToProtect(codec, targets)
-	targetToProtect := makeTargetToProtect(targets)
+	targetToProtect := makeCombinedTargetToProtect(targets)
 
 	log.VEventf(ctx, 2, "creating protected timestamp %v at %v", ptsID, resolved)
 	return jobsprotectedts.MakeRecord(
 		ptsID, int64(jobID), resolved, deprecatedSpansToProtect,
 		jobsprotectedts.Jobs, targetToProtect)
 }
 
+// createSystemTablesProtectedTimestampRecord will create a record to
+// protect the system tables at the resolved timestamp. We use this to protect
+// system tables separately from the user tables when per-table protected
+// timestamps are enabled to avoid duplicating the system tables protections
+// in each of the user tables' protected timestamp records.
+func createSystemTablesProtectedTimestampRecord(
+	ctx context.Context, codec keys.SQLCodec, jobID jobspb.JobID, resolved hlc.Timestamp,
+) *ptpb.Record {
+	ptsID := uuid.MakeV4()
+
+	log.VEventf(ctx, 2, "creating protected timestamp for system tables %v at %v", ptsID, resolved)
+	return jobsprotectedts.MakeRecord(
+		ptsID, int64(jobID), resolved, nil, jobsprotectedts.Jobs, makeSystemTablesTargetToProtect(),
```

Comment:
nit: add comment for the `nil` param name https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/181371303/Go+Golang+coding+guidelines#Avoid-Naked-Parameters

ditto for `createUserTablesProtectedTimestampRecord` below